services/user/Supervisor/ui/src/oauth/db.ts

```diff
@@ -0,0 +1,65 @@
+import { Result } from '../hostInterface';
+import { RecoverableErrorPayload } from '../plugin/errors';
+import { OAUTH_REQUEST_KEY, OAUTH_REQUEST_EXPIRATION } from '../constants';
+import { getSupervisor } from "@psibase/common-lib";
+
+export const supervisor = getSupervisor();
+export interface OauthRequest {
+    id: string,
+    subdomain: string,
+    subpath: string,
+    expiry_timestamp: number,
+}
+
+export const isTypeOauthRequest = (
+    oauthReqRes: OauthRequest | RecoverableErrorPayload,
+): oauthReqRes is OauthRequest => {
+    return (
+        "subdomain" in oauthReqRes &&
+        "subpath" in oauthReqRes &&
+        "expiry_timestamp" in oauthReqRes
+    );
+};
```

Comment:
There's a lot of `as` use here, wondering if it's worth using zod
```suggestion
const zOauthRequest = z.object({
    id: z.string(),
    subdomain: z.string(),
    subpath: z.string(),
    expiry_timestamp: z.number(),
})

export const isTypeOauthRequest = (
    oauthReqRes: OauthRequest | RecoverableErrorPayload,
): oauthReqRes is OauthRequest => {
    return zOauthRequest.safeParse(oauthReqRes).success
};
```

Reply:
Done. Thanks for spearheading our zodification @Velua